lib/Widgets/SettingsPage/AbstractSettingsPage.vala

```diff
@@ -21,15 +21,59 @@
  * AbstractSettingsPage is a #Gtk.ScrolledWindow subclass with properties used
  * by other Granite settings widgets.
  */
-
 public abstract class Granite.SettingsPage : Gtk.ScrolledWindow {
+    protected string _icon_name;
+    protected string _title;
+
+    public enum StatusType {
```

Comment:
I feel like this enum should have some kind of `to_icon_name` method so that it can be easily converted into a string that can be used in Gtk widgets.